http-client/src/main/java/io/airlift/http/client/jetty/JettyHttpClient.java

```diff
@@ -145,7 +158,14 @@ private JettyHttpClient(HttpClientConfig config, Optional<JettyIoPool> jettyIoPo
             sslContextFactory.setKeyStorePassword(config.getKeyStorePassword());
         }
 
-        httpClient = new HttpClient(new HttpClientTransportOverHTTP(2), sslContextFactory);
+        if (authenticationEnabled) {
+            requireNonNull(kerberosConfig.getConfig(), "kerberos config path is null");
```

Comment:
kerberosConfig is not, but kerberosConfig.getConfig() might be. It's only required to be non-null if auth is enabled